tests/v2/validation/upgrade/kubernetes_test.go

```diff
@@ -110,8 +111,11 @@ func (u *UpgradeKubernetesTestSuite) testUpgradeSingleCluster(clusterName, versi
 		u.T().Logf("[%v]: Validating updated cluster's nodepools kubernetes versions", clusterName)
 		validateNodepoolVersions(u.T(), client, updatedCluster, version, !isCheckingCurrentCluster)
 	}
-
-	err = nodestat.IsNodeReady(client, clusterMeta.ID)
+	if strings.Contains(versionToUpgrade, "rancher") {
+		err = nodestat.IsEveryRKE1NodeReady(client, clusterMeta.ID)
+	} else {
+		err = nodestat.IsEveryMachineReady(client, clusterMeta.ID)
+	}
```

Comment:
@caliskanugur is this fine?

Reply:
This might not be fine with this version. Having an else-if statement should make it more flexible. Currently, hosted clusters would fall under the `else` part. @markusewalker, what's your opinion on this?

Reply:
I think the else if just makes it a bit safer with respect to Ugur's comment on the hosted clusters. Probably better to change it now as opposed to later.

Reply:
I think this still needs to be addressed.

Reply:
is this addressed? not sure I'm following

Reply:
This is good to go?